docs/reference/mapping/dynamic/field-mapping.asciidoc

```diff
@@ -1,35 +1,36 @@
 [[dynamic-field-mapping]]
 === Dynamic field mapping
 
-By default, when a previously unseen field is found in a document,
-Elasticsearch will add the new field to the type mapping.   This behaviour can
-be disabled, both at the document and at the <<object,`object`>>  level, by
-setting the <<dynamic,`dynamic`>> parameter to `false` (to ignore new fields) or to `strict` (to throw
-an exception if an unknown field is encountered).
+When {es} detects a new field in a document, it _dynamically_ adds the field to
+the type mapping. The <<dynamic,`dynamic`>> parameter controls this behavior.
 
-Assuming `dynamic` field mapping is enabled, some simple rules are used to
-determine which data type the field should have:
+You can disable dynamic mapping, both at the document and at the
+<<object,`object`>> level. Setting the `dynamic` parameter to
+`false` ignores new fields, and `strict` throws an exception if {es} encounters
+an unknown field.
+
+When dynamic field mapping is enabled, {es} uses the following rules to
+determine how to map data types for each field.
 
 [horizontal]
-*JSON data type*::                   *Elasticsearch data type*
+*JSON data type*::                  *Elasticsearch data type*
 
 `null`::                            No field is added.
 `true` or `false`::                 <<boolean,`boolean`>> field
-floating{nbsp}point{nbsp}number::   <<number,`float`>> field
+floating{nbsp}point{nbsp}number::   <<number,`float`>> indexed field or <<number,`double`>> runtime field
 integer::                           <<number,`long`>> field
 object::                            <<object,`object`>> field
++
+NOTE: Objects are always mapped as part of the `properties` section, even when the `dynamic` parameter is set to `runtime`.
+
 array::                             Depends on the first non-`null` value in the array.
-string::                            Either a <<date,`date`>> field
-                                        (if the value passes <<date-detection,date detection>>),
-                                    a <<number,`double`>> or <<number,`long`>> field
-                                        (if the value passes <<numeric-detection,numeric detection>>)
-                                    or a <<text,`text`>> field, with a <<keyword,`keyword`>> sub-field.
+string::                            Either a <<date,`date`>> field (if the value passes <<date-detection,date detection>>), a <<number,`double`>> or <<number,`long`>> field (if the value passes <<numeric-detection,numeric detection>>), or a <<text,`text`>> field with a <<keyword,`keyword`>> sub-field.
```

Comment:
The probably should stay in a funny table format because its an "if, elsif, else" thing. Even worse now that we do different things if `dynamic: runtime`. Maybe this whole thing should turn into a big table with `dynamic:true` in one heading and `dynamic:runtime` in another? I dunno.

Reply:
After we chatted briefly, I agree that a tri-column table is the better way to present this information.